docs/COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,245 @@
+Commit Guidelines
+=================
+
+We enforce certain rules on commits with the following goals in mind:
+
+- Be able to reliably auto-generate the `CHANGELOG.md` *without* any human
+intervention.
+- Be able to automatically and correctly increment the semver version number
+based on what was done since the last release.
+- Be able to get a quick overview of what happened to the project by glancing
+over the commit history.
+- Be able to automatically reference relevant changes from a dependency
+upgrade.
+
+The guidelines are inspired by the [AngularJS git commit guidelines][angular-commit-guidelines].
+
+Commit structure
+----------------
+
+Each commit message consists of a header, a body and a footer. The header has a
+special format that includes a type, a scope and a subject.
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The subject should not contain more than 70 characters, including the type and
+scope, and the body should be wrapped at 72 characters.
+
+Type
+----
+
+Must be one of the following:
+
+- `feat`: A new feature.
+- `fix`: A bug fix.
+- `docs`: Documentation only changes.
+- `style`: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
+- `refactor`: A code change that neither fixes a bug nor adds a feature.
+- `perf`: A code change that improves performance.
+- `test`: Adding missing tests.
+- `chore`: Changes to the build process or auxiliary tools and libraries.
+- `upgrade`: A version upgrade of a project dependency.
+
+
+Scope
+-----
+
+The scope is required for types that make sense, such as `feat`, `fix`,
+`test`, etc. Certain commit types, such as `chore` might not have a clearly
+defined scope, in which case its better to omit it.
+
+When it applies, the scope must be either `GUI` or `CLI`.
+
+A commit that takes part in both the GUI and CLI scopes of the code should
+probably be split in commits that apply to each scope separately.
```

Comment:
Yeah maybe. At least in Etcher, the CLI and GUI code are very decoupled, and I very rarely encountered a case where such changes would make more sense together, but I can definitely see that happening in some cases. I'll fix the paragraph.